src/Material/Footer.elm

```diff
@@ -0,0 +1,590 @@
+  in
+    Options.styled elem
+      styles'
+      (List.map (contentToHtml tp) content)
+
+
+contentToHtml : Type -> Content a -> Html a
+contentToHtml tp content =
+  case content of
+    HtmlContent html ->
+      html
+
+    Content c ->
+      (toHtml tp c)
+
+
+sectionContent : Type -> String -> Content m -> Html m
+sectionContent tp section content =
+  let
+    pref =
+      prefix tp
+
+    sep =
+      separator
+  in
+    case content of
+      HtmlContent html ->
+        Options.styled Html.div
+          (cs (pref ++ sep ++ section) :: [])
+          [ html ]
+
+      Content { styles, content, elem } ->
+        Options.styled elem
+          (cs (pref ++ sep ++ section) :: styles)
+          (List.map (contentToHtml tp) content)
+
+
+dropdownHtml : Dropdown m -> Html m
+dropdownHtml (Dropdown { props, content }) =
+  Options.styled Html.div
+    (cs "mdl-mega-footer__drop-down-section" :: props)
+    (List.map (contentToHtml Mega) content)
+
+
+leftHtml : Type -> Maybe (Section a) -> List (Html a)
+leftHtml tp left =
+  case left of
+    Just (Section content) ->
+      [ sectionContent tp "left-section" content ]
+
+    Nothing ->
+      []
+
+
+rightHtml : Type -> Maybe (Section a) -> List (Html a)
+rightHtml tp right =
+  case right of
+    Just (Section content) ->
+      [ sectionContent tp "right-section" content ]
+
+    Nothing ->
+      []
+
+
+
+-- END OF INTERNAL HELPERS
+
+
+socialBtn : Property m
+socialBtn =
+  prefixedClass "social-btn"
+
+
+{-| Creates a footer logo
+-}
+logo : List (Property m) -> List (Content m) -> Content m
+logo styles content =
+  Content
+    { styles = (cs "mdl-logo" :: styles)
+    , content = content
+    , elem = Html.div
+    }
+
+
+{-| Creates a `link-list`
+-}
+links : List (Property m) -> List (Content m) -> Content m
+links styles content =
+  Content
+    { styles = (prefixedClass "link-list" :: styles)
+    , content = content
+    , elem = Html.ul
+    }
+
+
+{-| Creates a link
+-}
+link : List (Property m) -> List (Html m) -> Content m
+link styles contents =
+  wrap
+    <| Options.styled a
+        styles
+        contents
+
+
+li : List (Property m) -> List (Html m) -> Content m
+li styles content =
+  wrap
+    <| Options.styled Html.li
+        styles
+        content
+
+
+{-| Creates a link wrapped in a `li`-element
+-}
+linkItem : List (Property m) -> List (Content m) -> Content m
+linkItem styles content =
+  Content
+    { styles = []
+    , content = [ Content { styles = styles, content = content, elem = Html.a } ]
+    , elem = Html.li
+    }
+
+
+headingClass : Property m
+headingClass =
+  prefixedClass "heading"
+
+
+{-| Creates a footer `heading` element
+-}
+heading : List (Property m) -> List (Content m) -> Content m
+heading styles content =
+  Content
+    { styles = headingClass :: styles
+    , content = content
+    , elem = Html.h1
+    }
+
+
+{-| Wraps `Html.text` element to `Content`
+-}
+text : String -> Content m
```

Comment:
I think `text` is unnecessary here, `heading` I think might be okay, it is very specifically an h1 element with the class `mdl-mega-footer__heading`

> Defines a heading as a mega-footer heading	
> Required on h1 element inside drop-down section